main.py

```diff
@@ -430,3 +432,18 @@ def decode_html(text):
 plt.ylabel("Predicted Values")
 plt.title("Scatter Plot: Predicted vs Actual (Testing Data)")
 plt.show()
+
+
+##### Random Forest Algorithm with Random UnderSampler #####
+sampler = RandomUnderSampler()
+X_train1, y_train1 = sampler.fit_resample(X_train, Y_train)
+
+clf2= RandomForestClassifier().fit(X_train1, y_train1)
+
+#Make Predictions
+pred = clf.predict(X_test)
+
+# Evaluating
+print("Accuracy: %s" % str(clf.score(X_test, Y_test)))
+print("Confusion Matrix")
+print(confusion_matrix(pred, Y_test))
```

Comment:
The confusion matrix is computed using `confusion_matrix(pred, Y_test)`, but `pred` seems to be undefined. You should replace `pred` with `clf2.predict(X_test)`.